gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/scoring/use_case/SaveScoringResponseUseCase.java

```diff
@@ -68,17 +71,46 @@ public Completable execute(Input input) {
     }
 
     private ScoringReport.Summary processSummary(List<ScoringReport.Asset> assets) {
-        var summary = assets
-            .stream()
-            .flatMap(asset -> asset.diagnostics().stream())
-            .collect(Collectors.groupingBy(ScoringReport.Diagnostic::severity, Collectors.counting()));
+        var assetsSummary = new ArrayList<ScoringReport.Summary>();
+        for (var asset : assets) {
+            var counters = asset
+                .diagnostics()
+                .stream()
+                .collect(Collectors.groupingBy(ScoringReport.Diagnostic::severity, Collectors.counting()));
+
+            var nbErrors = counters.getOrDefault(ScoringReport.Severity.ERROR, 0L);
+            var nbWarnings = counters.getOrDefault(ScoringReport.Severity.WARN, 0L);
+            var nbInfos = counters.getOrDefault(ScoringReport.Severity.INFO, 0L);
+            var nbHints = counters.getOrDefault(ScoringReport.Severity.HINT, 0L);
 
-        return new ScoringReport.Summary(
-            summary.getOrDefault(ScoringReport.Severity.ERROR, 0L),
-            summary.getOrDefault(ScoringReport.Severity.WARN, 0L),
-            summary.getOrDefault(ScoringReport.Severity.INFO, 0L),
-            summary.getOrDefault(ScoringReport.Severity.HINT, 0L)
-        );
+            assetsSummary.add(
+                new ScoringReport.Summary(
+                    scoreComputingDomainService.computeScore(nbErrors, nbWarnings, nbInfos, nbHints),
+                    nbErrors,
+                    nbWarnings,
+                    nbInfos,
+                    nbHints
+                )
+            );
+        }
+
+        var averageScore = BigDecimal
+            .valueOf(assetsSummary.stream().mapToDouble(ScoringReport.Summary::score).average().orElse(0.0))
```

Comment:
The mapToDouble() will throw exception if score is null should we filter this? evaluate to 0? or we know that null is impossible and maybe use double instead of Double?